packages/account/src/Sections/Verification/ProofOfOwnership/proof-of-ownership-form.jsx

```diff
@@ -22,109 +22,150 @@ const ProofOfOwnershipForm = ({
     refreshNotifications,
-    total_documents_required,
     updateAccountStatus,
 }) => {
     const grouped_payment_method_data_keys = Object.keys(grouped_payment_method_data);
     const initial_values = {};
-    const [is_submit_button_disabled, setIsSubmitButtonDisabled] = React.useState(true);
     const [form_state, setFormState] = useStateCallback({ should_show_form: true });
-    const form_ref = useRef();
-    const [document_upload_errors, setDocumentUploadErrors] = useState();
-    useEffect(() => {
-        setDocumentUploadErrors([]);
-    }, []);
+    const form_ref = React.useRef();
     const fileReadErrorMessage = filename => {
         return localize('Unable to read file {{name}}', { name: filename });
     };
-
     const validateFields = values => {
-        const errors = {};
-        errors.data = [...document_upload_errors];
+        let errors = {};
+        errors.data = [...(form_ref?.current?.errors?.data || [])];
         let total_documents_uploaded = 0;
         let has_errors = false;
+        let are_files_uploaded = false;
         const cards = values?.data;
         Object.keys(cards)?.forEach?.(card_key => {
             const items = cards?.[card_key] ?? {};
-            errors.data[card_key] = errors.data?.[card_key] ?? [];
-            Object.keys(items)?.forEach?.(item_key => {
-                errors.data[card_key][item_key] = errors.data?.[card_key]?.[item_key] ?? {};
-                const payment_method = items?.[item_key];
-                const payment_method_identifier = payment_method?.payment_method_identifier?.trim();
-                const is_pm_identifier_provided =
-                    payment_method?.is_generic_pm || payment_method_identifier?.length > 0;
-                const is_credit_or_debit_card = payment_method?.identifier_type === IDENTIFIER_TYPES.card_number;
-                total_documents_uploaded += payment_method?.files?.filter(Boolean)?.length ?? 0;
-                payment_method?.files?.forEach((file, i) => {
-                    errors.data[card_key][item_key].files = errors?.data?.[card_key]?.[item_key]?.files ?? [];
-                    if (file?.type && !/(image|application)\/(jpe?g|pdf|png)$/.test(file?.type)) {
-                        errors.data[card_key][item_key].files[i] = localize(
-                            "That file format isn't supported. Please upload .pdf, .png, .jpg, or .jpeg files only."
-                        );
+            const item_keys = Object.keys(items);
+            item_keys?.forEach?.(item_key => {
+                if (!has_errors) {
+                    errors.data[card_key] = errors.data?.[card_key] ?? {};
+                    errors.data[card_key][item_key] = errors.data?.[card_key]?.[item_key] ?? {};
+                    const payment_method = items?.[item_key];
+                    const payment_method_identifier = payment_method?.payment_method_identifier?.trim();
+                    const is_pm_identifier_provided =
```

Comment:
nit
```suggestion
                    const is_payment_method_identifier_provided =
```

Reply:
@shayan-deriv for the `errors.data[card_key][item_key]` and `form_ref.current?.values?.data?.[card_key]`. Can you please help with the meaningful name? 😃

Reply:
@jim-deriv sure man. let's have a call and check it. because I'm not sure if I understood what exactly they are for